src/native/mqtt_connection.c

```diff
@@ -202,7 +215,8 @@ static void s_on_connection_interrupted_internal(
     int error_code,
     jobject ack_callback) {
 
-    JNIEnv *env = aws_jni_get_thread_env(connection->jvm);
+    /********** JNI ENV ACQUIRE **********/
+    JNIEnv *env = aws_jni_acquire_thread_env(connection->jvm);
```

Comment:
`s_on_connection_interrupted_internal()` is a helper function, not a callback, we could just pass `env` in as an argument